posthog/api/test/test_capture.py

```diff
@@ -143,6 +162,148 @@ def test_capture_event(self, kafka_produce):
             self._to_arguments(kafka_produce),
         )
 
+    @patch("posthog.kafka_client.client._KafkaProducer.produce")
+    def test_capture_event_with_db_error_no_default(self, kafka_produce):
+        """Test database errors while fetching instance settings don't fail capture.
+
+        Even though we are creating an InstanceSetting, it should not be fetched as we
+        force a database error to occur (with the DatabaseErrorWrapper). This should
+        cause us to use the default EVENT_PARTITION_KEYS_TO_OVERRIDE setting. As this
+        is not set to the current expected_key, no override will occur.
+        """
+        distinct_id = "999"
+        expected_key = f"{self.team.pk}:{distinct_id}"
+        key = CONSTANCE_DATABASE_PREFIX + "EVENT_PARTITION_KEYS_TO_OVERRIDE"
+
+        InstanceSetting.objects.create(key=key, raw_value=f'["{expected_key}"]')
```

Comment:
Rather than using the model implementation detail here, could we use `set_instance_setting` instead to allow us to change the implementation without needing to update these tests?

Reply:
Fair enough, I'll make the change.